### DISCUSSION TITLE
How fast a farmer could possibly be?

### DISCUSSION THREAD

**Discussion by iliasorazov** [Chia-Network/chia-blockchain#7046] [Support]
I am wondering how slow a harvester and a farmer can be in submitting the proof to successfully mine a block? The code from `harvester_api.py` issue warning if a quality lookup took more than 5 seconds. But then I open the Chia Network Consensus paper on infusion point and timelords and there I read:  

> the infusion iterations will be between 3 and 4 signage points after the signage point. Farmers must submit their proofs and blocks before the infusion point is reached 

and also

> After realizing they have won (at the 14th infusion point), the farmer fetches the whole proof of space, makes a block, optionally including transactions, and broadcasts this to the network. They have a few seconds (up to the infusion iterations), to reach timelords. 

But 3-4 signage point delay is something around 30 seconds at the current network rate and it is far more than a "few seconds". This paper is very complicated to understand. So what is the actual latency a farmers must have to mine a block? How long can possibly be a time interval between a signage point challenge hash generation and the time when the farmer published the quality and the POS to the network?

**Comment by loppefaaret** [comment]
the above papers are correct - given a single instance, you could get close to 30 seconds, and still win - but if you consider that there is a signage point about every 9.375 seconds by design, if you spend 25 seconds processing one signage point, you are already close to 15 seconds behind the following signage point
i would also assume that it is not unlikely for 4 consecutive signage points to be "fast" 5 second rounds, thus forcing you to process all of them in less than 5 seconds each to keep up - hence why we are WARNING users already at 5 seconds scan time!

**Reply by iliasorazov** [reply]
Thanks for the reply! It is latency not bandwidth, plots are read asynchronous and in parallel. It is very unlikely that the same plot or even the same harvester would have a block winning POS for consequent signage points. So even if they came fast, 5 seconds each, I would still have 15-20 seconds to complete the full proof read operation on the first plot based on the first signage point and safely broadcast it to the network, right?

**Reply by loppefaaret** [reply]
that would be a fair assumption in my opinion

**Comment by BrandtH22** [comment]
Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
